homeassistant/components/aussie_broadband/config_flow.py

```diff
@@ -0,0 +1,184 @@
+"""Config flow for Aussie Broadband integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiohttp import ClientError
+from aussiebb.asyncio import AussieBB, AuthenticationException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_SERVICES, DOMAIN, SERVICE_ID
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aussie Broadband."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.data: dict = {}
+        self.options: dict = {CONF_SERVICES: []}
+        self.services: list[dict[str]] = []
+        self.client: AussieBB | None = None
+        self._reauth_username: str | None = None
+
+    async def async_auth(self, user_input: dict[str, str]) -> dict[str, str] | None:
+        """Reusable Auth Helper."""
+        self.client = AussieBB(
+            user_input[CONF_USERNAME],
+            user_input[CONF_PASSWORD],
+            async_get_clientsession(self.hass),
+        )
+        try:
+            await self.client.login()
+            return None
+        except AuthenticationException:
+            return {"base": "invalid_auth"}
+        except ClientError:
+            return {"base": "cannot_connect"}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] | None = None
+        if user_input is not None:
+            if not (errors := await self.async_auth(user_input)):
+                await self.async_set_unique_id(user_input[CONF_USERNAME])
```

Comment:
Does the username need to be normalized here?

https://developers.home-assistant.io/docs/config_entries_config_flow_handler?_highlight=unique&_highlight=id#sometimes-acceptable-sources-for-a-unique-id-for-cloud-services

Reply:
I just tried it and confirmed that the username is case sensitive so should not be normalized.

Reply:
This is done now. both username and reauth username are lowered when calling async_set_unique_id